packages/eas-cli/src/user/expoSsoLauncher.ts

```diff
@@ -0,0 +1,82 @@
+import openBrowserAsync from 'better-opn';
+import http from 'http';
+import { Socket } from 'node:net';
+import querystring from 'querystring';
+
+import Log from '../log';
+
+export default (options: {
+  expoWebsiteUrl: string;
+  serverPort: number;
+}): { executeAuthFlow: () => Promise<string> } => {
+  const { expoWebsiteUrl, serverPort } = options;
+  if (!expoWebsiteUrl || !serverPort) {
+    throw new Error('Expo website URL and localserver port are required.');
+  }
+  const scheme = 'http';
+  const hostname = 'localhost';
+  const path = '/auth/callback';
+  const redirectUri = `${scheme}://${hostname}:${serverPort}${path}`;
+
+  const buildExpoSsoLoginUrl = (): string => {
+    const data = {
+      app_redirect_uri: redirectUri,
+    };
+    const params = querystring.stringify(data);
+    return `${expoWebsiteUrl}/sso-login?${params}`;
+  };
+
+  // Start server and begin auth flow
+  const executeAuthFlow = (): Promise<string> => {
+    return new Promise<string>(async (resolve, reject) => {
+      const connections = new Set<Socket>();
+
+      const server = http.createServer(
+        (request: http.IncomingMessage, response: http.ServerResponse) => {
+          try {
+            if (!(request.method === 'GET' && request.url?.includes('/auth/callback'))) {
+              throw new Error('Unexpected SSO login response');
+            }
+            const url = new URL(request.url, `http:${request.headers.host}`);
```

Comment:
I think the following should work? (so we don't need the odd http thing)

```suggestion
            const url = new URL(request.url);
```